src/main/java/com/twilio/taskrouter/application/servlet/EventsServlet.java

```diff
@@ -25,48 +31,85 @@
 @Singleton
 public class EventsServlet extends HttpServlet {
 
-  public static final String LEAVE_MSG = "Sorry, All agents are busy. Please leave a message. "
+  protected static final String LEAVE_MSG = "Sorry, All agents are busy. Please leave a message. "
     + "We will call you as soon as possible";
 
+  protected static final String OFFLINE_MSG = "Your status has changed to Offline. "
+    + "Reply with \"On\" to get back Online";
+
   private static final Logger LOG = Logger.getLogger(EventsServlet.class.getName());
 
   private final TwilioAppSettings twilioSettings;
 
   private final MissedCallRepository missedCallRepository;
 
   @Inject
-  public EventsServlet(final TwilioAppSettings twilioSettings,
-                       final MissedCallRepository missedCallRepository) {
+  public EventsServlet(TwilioAppSettings twilioSettings,
+                       MissedCallRepository missedCallRepository) {
     this.twilioSettings = twilioSettings;
     this.missedCallRepository = missedCallRepository;
   }
 
   @Override
   public void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException,
     IOException {
-    Optional.ofNullable(req.getParameter(TwilioAppSettings.EVENT_TYPE_PARAM))
-      .filter(TwilioAppSettings.DESIRABLE_EVENTS::contains)
-      .flatMap(eventType -> getTaskAttributes(req)).ifPresent(this::addMissingCallAndLeaveMessage);
+    Optional.ofNullable(req.getParameter("EventType"))
+      .ifPresent(eventName -> {
+        switch (eventName) {
+          case "workflow.timeout":
+          case "task.canceled":
+            parseAttributes("TaskAttributes", req)
+              .ifPresent(this::addMissingCallAndLeaveMessage);
+            break;
+          case "worker.activity.update":
+            Optional.ofNullable(req.getParameter("WorkerActivityName"))
+              .filter("Offline"::equals)
+              .ifPresent(offlineEvent -> {
+                parseAttributes("WorkerAttributes", req)
+                  .ifPresent(this::notifyOfflineStatusToWorker);
+              });
+            break;
+          default:
+        }
+      });
+  }
+
+  protected Optional<JsonObject> parseAttributes(String parameter, HttpServletRequest request) {
```

Comment:
ditto. Make sure we're using the right attributes accessors. `protected` should be used only if we are intended to use inheritance and make the attributes or methods available for the extended class.

Please make the appropriate changes.